src/wp-admin/edit-form-blocks.php

```diff
@@ -57,44 +60,10 @@
 	sprintf( '/wp/v2/%s/%d/autosaves?context=edit', $rest_base, $post->ID ),
 );
 
-/**
- * Preload common data by specifying an array of REST API paths that will be preloaded.
- *
- * Filters the array of paths that will be preloaded.
- *
- * @since 5.0.0
- *
- * @param string[] $preload_paths Array of paths to preload.
- * @param WP_Post  $post          Post being edited.
- */
-$preload_paths = apply_filters( 'block_editor_preload_paths', $preload_paths, $post );
-
-/*
- * Ensure the global $post remains the same after API data is preloaded.
- * Because API preloading can call the_content and other filters, plugins
- * can unexpectedly modify $post.
- */
-$backup_global_post = clone $post;
-
-$preload_data = array_reduce(
-	$preload_paths,
-	'rest_preload_api_request',
-	array()
-);
-
-// Restore the global $post as it was before API preloading.
-$post = $backup_global_post;
-
-wp_add_inline_script(
-	'wp-api-fetch',
-	sprintf( 'wp.apiFetch.use( wp.apiFetch.createPreloadingMiddleware( %s ) );', wp_json_encode( $preload_data ) ),
-	'after'
-);
-
-wp_add_inline_script(
-	'wp-blocks',
-	sprintf( 'wp.blocks.setCategories( %s );', wp_json_encode( get_block_categories( $post ) ) ),
-	'after'
+block_editor_configure(
```

Comment:
This looks a bit "sudden" when reading the file without the removed / moved code so I'd add a comment about what the call to this does (add scripts etc.).

Reply:
Sure, we can copy and adjust the description of the function 😄

Reply:
An alternative I consider is to use more existing filters to do it in a way that works with all block editors automatically.

Reply:
I replaced this function with `block_editor_preload_api_requests` that is only concerned with setting the preloaded data for API requests.